src/sage/schemes/elliptic_curves/ell_rational_field.py

```diff
@@ -2218,13 +2289,18 @@ def gens(self, proof=None, **kwds):
             :meth:`~gens_certain` method to find out afterwards
             whether the generators were proved.
 
-        IMPLEMENTATION: Uses Cremona's mwrank C library.
+        IMPLEMENTATION: Uses Cremona's mwrank C++ library or ellrank in pari.
 
         EXAMPLES::
 
             sage: E = EllipticCurve('389a')
             sage: E.gens()                 # random output
             [(-1 : 1 : 1), (0 : 0 : 1)]
+            sage: E.gens(algorithm="pari")    # random output
+            [(5/4 : 5/8 : 1), (0 : 0 : 1)]
+            sage: E = EllipticCurve([0,2429469980725060,0,275130703388172136833647756388,0])
+            sage: len(E.gens(algorithm="pari"))
+            14
```

Comment:
This test takes 50s, and brings testing the whole file (around 900 tests) from 22s in sagemath-10.0 to 73s in sagemath-10.1.

This doesn't seem ok even for `--long`.

Reply:
This looks like an example where the old default (mwrank) is better than the newly implemented pari code.

Reply:
Indeed, mwrank takes 13s while pari takes 51s. The default seems to be still mwrank.

In any case, I would suggest that this test be marked "# not tested, too long" or something (or be replaced by a smaller rank example which takes <1s).

Reply:
Fully agree. It should stay in the documentation as a complicated example, but not be part of the tests. (Just back from holidays and fairly busy for a while) Please feel free to change this, I will do it but it will take a while.